src/solvers/flattening/boolbv.cpp

```diff
@@ -85,8 +85,11 @@ bvt boolbvt::conversion_failed(const exprt &expr)
   ignoring(expr);
 
   // try to make it free bits
-  std::size_t width=boolbv_width(expr.type());
-  return prop.new_variables(width);
+  // TODO we likely end up in this path when an earlier call to boolbv_width
```

Comment:
Does the TODO mean that something needs to be done or that we don't understand why this piece of code exists?

Reply:
Thank you for reminding me of this 😊 I have now audited all uses of `conversion_failed`, several of which were in response to calls to `boolbv_width` failing. That, however, would have resulted in the call to `boolbv_width` in here failing, making the code path really dangerous in the way it previously was (we would not have generated the correct number of free variables), and useless the way I changed it (making the `CHECK_RETURN` in here fail). All of these instances are now turned into early failures.